clients/web/test/spec/utilitiesSpec.js

```diff
@@ -0,0 +1,44 @@
+girderTest.startApp();
+
+describe('Test EventStream', function () {
+    it('test heartbeat', function () {
+        var es = girder.utilities.eventStream,
+            start = 0,
+            stop = 0,
+            close = 0;
+        runs(function () {
+            es.on('g:eventStream.stop', function () {
+                stop += 1;
+            });
+            es.on('g:eventStream.start', function () {
+                start += 1;
+            });
+            es.on('g:eventStream.close', function () {
+                close += 1;
+            });
+            es.settings._heartbeatTimeout = 1;
+            start = stop = close = 0;
+            es.open();
+        }, 'start event stream and set it to have a short timeout');
+        waitsFor(function () {
+            return stop;
+        }, 'event stream heartbest to stop');
+        runs(function () {
+            expect(stop).toBeGreaterThan(0);
+            es.settings._heartbeatTimeout = 5000;
+        }, 'restart heartbeat');
+        waitsFor(function () {
+            return start;
+        }, 'restart to occur');
+        runs(function () {
+            expect(start).toBeGreaterThan(0);
```

Comment:
The counters are reinitialized on line 20.  Since we are using a short timeout to trigger the stop, it is possible the restart will occur before we reset the timeout to `5000` (since the `runs` command actually executes in the next timeslice in jasmine then the last `waits` command), so we don't want to reset the start counter again.